clippy_utils/src/str_utils.rs

```diff
@@ -289,6 +296,49 @@ pub fn to_camel_case(item_name: &str) -> String {
     s
 }
 
+/// Return an hint if the `arg` is a string with a single character
+pub fn get_hint_if_single_char_arg(
+    cx: &LateContext<'_>,
+    arg: &Expr<'_>,
+    applicability: &mut Applicability,
+    ascii_only: bool,
+) -> Option<String> {
```

Comment:
Let's rename this and expand the description a bit

```suggestion
/// Converts `expr` to a `char` literal if it's a `str` literal containing a single character (or a single byte with `ascii_only`)
pub fn str_literal_to_char_literal(
    cx: &LateContext<'_>,
    expr: &Expr<'_>,
    applicability: &mut Applicability,
    ascii_only: bool,
) -> Option<String> {
```

Reply:
I think it belongs more in `clippy_utils::source` also, this module is for operations on the non syntax kind of `str`s

Reply:
Ok thanks for the explanations. Applied